### PR TITLE
GameEngine Part 1&2

### PR DESCRIPTION
**Pull request by Swess**
Contributors: @Swess & @Kerry-G

### CODE REVIEW

File: src/GameEngine/GameEngine.cpp [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        select_player();
+        Terminal::debug("Game Engine Done.");
+    }
+
+    GameEngine::GameEngine() {
+        map = nullptr;
+        deck = nullptr;
+        players = new vector<Player::Player *>();
+        player_order = new vector<int>();
+        // map_registry = new Map::MapRegistry();
```
Comment (Kerry-G) [inline src/GameEngine/GameEngine.cpp]: commented code can be remove [fixed]

File: src/GameEngine/GameEngine.cpp [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    void GameEngine::assign_country_to_player() {
+        // Loop countries with player index increasing
+        vector<Country *> countries = map->get_countries();
+
+        shuffle( countries.begin(), countries.end(), default_random_engine(time(0)));
+
+        int owner_index = 0;
+        for(auto & countrie : countries){
```
Comment (Kerry-G) [inline src/GameEngine/GameEngine.cpp]: countrie sounds cute, but it's spelled country 😄 - [fixed]

File: src/GameEngine/GameEngine.cpp [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+            players->at(owner_index)->gain_control(countrie);
+
+            Terminal::debug("Gave control of country '"+countrie->get_name()+"' to player "+to_string(owner_index) );
+            owner_index = ++owner_index % (int)players->size();
+        }
+    }
+
+    void GameEngine::assign_armies_into_country() {
+        int total_placed = 0;
+        int armies_allowed = 40 - (players->size()-2)*5;
```
Comment (Kerry-G) [inline src/GameEngine/GameEngine.cpp:237]: nice

**Review by Kerry-G — COMMENTED** [review]
[redacted]

File: src/GameEngine/GameEngine.cpp (hunk 2 above)
Comment (Swess) [reply src/GameEngine/GameEngine.cpp]: Lol IDE auto-completed I guess XD thx

**Review by Kerry-G — APPROVED** [review]

**Pull request merged**